src/sagemaker/jumpstart/payload_utils.py

```diff
@@ -0,0 +1,134 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""This module stores payload utilities for SageMaker JumpStart."""
+from __future__ import absolute_import
+import base64
+import json
+from typing import Any, Optional, Union
+import re
+import boto3
+
+from sagemaker.jumpstart.accessors import JumpStartS3Accessor
+from sagemaker.jumpstart.constants import JUMPSTART_DEFAULT_REGION_NAME
+from sagemaker.jumpstart.enums import MIMEType
+from sagemaker.jumpstart.types import JumpStartSerializablePayload
+from sagemaker.jumpstart.utils import get_jumpstart_content_bucket
+
+S3_BYTES_REGEX = r"^\$s3<(?P<s3_key>[a-zA-Z0-9-_/.]+)>$"
+S3_B64_STR_REGEX = r"\$s3_b64<(?P<s3_key>[a-zA-Z0-9-_/.]+)>"
+
+
+class PayloadSerializer:
+    """Utility class for serializing payloads associated with JumpStart models.
+
+    Many JumpStart models embed byte-streams into payloads corresponding to images, sounds,
+    and other content types which require downloading from S3.
+    """
+
+    def __init__(
+        self,
+        bucket: str = get_jumpstart_content_bucket(),
+        region: str = JUMPSTART_DEFAULT_REGION_NAME,
+        s3_client: Optional[boto3.client] = None,
+    ) -> None:
+        """Initializes PayloadSerializer object."""
+        self.bucket = bucket
+        self.region = region
+        self.s3_client = s3_client
+
+    def get_bytes_payload_with_s3_references(
+        self,
+        payload_str: str,
+    ) -> bytes:
+        """Returns bytes object corresponding to referenced s3 object.
+
+        Raises:
+            ValueError: If the raw bytes payload is not formatted correctly.
+        """
+        s3_keys = re.compile(S3_BYTES_REGEX).findall(payload_str)
+        if len(s3_keys) != 1:
+            raise ValueError(f"Invalid bytes payload: {payload_str}")
+
+        s3_key = s3_keys[0]
+        serialized_s3_object = JumpStartS3Accessor.get_object_cached(
```

Comment:
caching this, are you sure?

Reply:
Yes, so that if you have multiple payloads that share an underlying s3 artifact, you only query s3 once.